tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -5659,15 +5656,9 @@ class HLSLExternalSource : public ExternalSemaSource {
       CXXRecordDecl *Decl = arg.getAsType()->getAsCXXRecordDecl();
       if (Decl && !Decl->isCompleteDefinition())
         return true;
-      if (ContainsLongVector(arg.getAsType())) {
-        const unsigned TessellationPatchesIDx = 1;
-        m_sema->Diag(argLoc.getLocation(),
-                     diag::err_hlsl_unsupported_long_vector)
-            << TessellationPatchesIDx;
-        return true;
-      }
+      const TypeDiagContext DiagContext = TypeDiagContext::TessellationPatches;
       if (DiagnoseTypeElements(*m_sema, argLoc.getLocation(), arg.getAsType(),
-                               TypeDiagContext::TessellationPatches))
+                               DiagContext, DiagContext))
```

Comment:
As frequently as the pattern of both objects and long vectors having the same context occurs, it may make sense to have an overload of `DiagnoseTypeElements` that takes a single context that is propagated to both for the longer overload, but I won't insist on it if there are no other commits to be made to this PR that change might be packaged with.